sotodlib/preprocess/processes.py

```diff
@@ -779,6 +779,50 @@ def process(self, aman, proc_aman):
                 subtract_name = self.process_cfgs["subtract_name"]
                 )
 
+class A2Stats(_Preprocess):
+    """
+    Calculate statistical metrics for A2, the 2f-demodulated Q and U signals.
+
+    Example config block::
+
+      - name: "a2_stats"
+        calc:
+          stat_names: ["mean", "median", "var", "ptp"]
+          subscan: True
+        save: True
+
+    """
+    name = "a2_stats"
+
+    def calc_and_save(self, aman, proc_aman):
+        # Get A2 signal using the demod_tod function; this function generates intermediary
+        # dsT, demodQ, demodU fields in the AxisManager that will be deleted
+        hwp.demod_tod(aman, demod_mode=2)
```

Comment:
I think modify hwp.hwp.demod_tod to have a `wrap` argument that defaults to True lines 643-657 would then be wrapped in a `if wrap:` block and the `else` would just return the 3 arrays. Then here you could just return the 3 arrays and not worry about wrapping them and deleting them from `aman` later.